### PR TITLE
Beheka hw 8

### PR DESCRIPTION
**Pull request by MishaBeheka**

### CODE REVIEW

File: src/main/java/core/basesyntax/MyLinkedList.java [hunk 1]
```diff
@@ -3,48 +3,132 @@
...
+        willBeRemove.next.prev = willBeRemove.prev;
+        size--;
+        return willBeRemove.value;
     }
 
     @Override
     public T remove(T t) {
+        Node<T> willBeRemove = head;
+        for (int i = 0; i < size; i++) {
+            if (t != null && t.equals(willBeRemove.value)) {
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/MyLinkedList.java]: if `t` ==null? What will happen?

File: src/main/java/core/basesyntax/MyLinkedList.java [hunk 2]
```diff
@@ -3,48 +3,132 @@
...
+        }
+        Node<T> willBeRemove = returnNodeByIndex(index);
+        willBeRemove.prev.next = willBeRemove.next;
+        willBeRemove.next.prev = willBeRemove.prev;
+        size--;
+        return willBeRemove.value;
     }
 
     @Override
     public T remove(T t) {
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/MyLinkedList.java]: where do you remove the value?

Comment (MishaBeheka) [reply src/main/java/core/basesyntax/MyLinkedList.java]: I do not understand the question. Need to reset the link to this object?

File: src/main/java/core/basesyntax/MyLinkedList.java [hunk 3]
```diff
@@ -3,48 +3,132 @@
...
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (size == index) {
+            add(value);
+            return;
+        }
+        Node<T> clone = returnNodeByIndex(index);
```
[the remark this answers is not in the document]
Comment (MishaBeheka) [reply src/main/java/core/basesyntax/MyLinkedList.java]: Renamed

File: src/main/java/core/basesyntax/MyLinkedList.java [hunk 4]
```diff
@@ -3,48 +3,132 @@
...
+        clone.prev.next = newNode;
+        clone.prev = newNode;
+        size++;
 
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (T i : list) {
```
[the remark this answers is not in the document]
Comment (MishaBeheka) [reply src/main/java/core/basesyntax/MyLinkedList.java]: renamed

File: src/main/java/core/basesyntax/MyLinkedList.java (hunk 1 above)
Comment (MishaBeheka) [reply src/main/java/core/basesyntax/MyLinkedList.java]: add check t == null

File: src/main/java/core/basesyntax/MyLinkedList.java (hunk 2 above)
Comment (MishaBeheka) [reply src/main/java/core/basesyntax/MyLinkedList.java]: Added 
 willBeRemove.next = null;
 willBeRemove.prev = null;

Comment (MishaBeheka) [reply src/main/java/core/basesyntax/MyLinkedList.java]: Is that what you meant?

**Pull request closed without merging**